### ISSUE TITLE
Guix not working in complex commands

### ISSUE THREAD

**Issue by guilhermehas** [ethancedwards8/nixos-guix#2]
The simple command `guix build hello` is working in my machine. But other commands like `guix pull` or `guix build nix` don't work here. Have you tried to run this on your computer or VM? If in a VM, could you tell what you made for it works?

**Comment by ethancedwards8** [comment]
I've also experienced this. I haven't found a fix as of yet. Hopefully, a solution can be found.

**Comment by guilhermehas** [comment]
I tried to automate the creation of VM in my repository:
https://github.com/guilhermehas/nixos-guix-vm

One time, I got the bug running `guix build hello --no-substitutes`.
If you can simulate the bug in my VM, let me know.
I am trying to make this repository very easy to use, so let me know if I can make it easier to use.

I will send this repository for the developers when I can easily reproduce this bug in the VM.

**Comment by ethancedwards8** [comment]
> I tried to automate the creation of VM in my repository:
> https://github.com/guilhermehas/nixos-guix-vm
> 
> One time, I got the bug running `guix build hello --no-substitutes`.
> If you can simulate the bug in my VM, let me know.
> I am trying to make this repository very easy to use, so let me know if I can make it easier to use.
> 
> I will send this repository for the developers when I can easily reproduce this bug in the VM.

Okay thank you, I'll give it a try. Sorry for the inactivity the past few days, I've been on vacation.

Maybe we can integrate your VM into a test suite? That would make everything easier. But right now the VM looks good.

**Comment by guilhermehas** [comment]
Do you know how to put a VM into a test suite using nix flakes?
I just know how to do it by putting it inside nixpkgs repository.

**Comment by ethancedwards8** [comment]
> Do you know how to put a VM into a test suite using nix flakes?
> I just know how to do it by putting it inside nixpkgs repository.

Just add a flake output? Should be simple enough.

**Comment by guilhermehas** [comment]
Creating a new flake output is the easiest part. I was talking about using this https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/make-test-python.nix . The problem is that it is not an output of nixpkgs. And I can't copy and paste this file in my directory because it is referencing other files.

**Comment by ethancedwards8** [comment]
> Creating a new flake output is the easiest part. I was talking about using this https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/make-test-python.nix . The problem is that it is not an output of nixpkgs. And I can't copy and paste this file in my directory because it is referencing other files.

Oh I see what you mean, yeah hmm.

Anyway, using your repo is fine for right now. I think the current priority should be resolving the issue at hand. Do you have any leads on whats causing the failure?

**Comment by guilhermehas** [comment]
No. In the VM, there are several more errors that don't happen in my machine. And the error that happens in my machine doesn't happen in the VM.

**Comment by ethancedwards8** [comment]
> No. In the VM, there are several more errors that don't happen in my machine. And the error that happens in my machine doesn't happen in the VM.

Well thats odd. And not reproducible at all. I'll try to play with it more when I have some more free time.
